storm-core/test/clj/integration/org/apache/storm/integration_test.clj

```diff
@@ -403,17 +403,18 @@
 (def bolt-prepared? (atom false))
 (defbolt prepare-tracked-bolt [] {:prepare true}
   [conf context collector]  
-  (reset! bolt-prepared? true)
   (bolt
    (execute [tuple]
+            (reset! bolt-prepared? true)
```

Comment:
The basic idea was that, in the original code, `spout-opened?` or `bolt-prepared?` are set to true in executor initialization phase, while after porting to java, this behavior is deferred to `execute/nextTuple`.